src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -386,9 +389,10 @@ public void cleanup(CleanupStrategy strategy) {
           LOGGER.error(
               () -> String.format("Could not delete a local repository: %s", info.path()), e);
         }
-        localRepositoriesInfo.remove(url);
+        toBeRemoved.remove(url);
```

Comment:
Did you mean to add a URL to the set?

```suggestion
        toBeRemoved.add(url);
```

Reply:
🙈 Sure, this would make a lot more sense 🙂